pinot-query-planner/src/main/java/org/apache/pinot/query/rules/PinotExchangeNodeInsertRule.java

```diff
@@ -57,12 +63,26 @@ public boolean matches(RelOptRuleCall call) {
 
   @Override
   public void onMatch(RelOptRuleCall call) {
+    // TODO: this only works for single equality JOIN. add generic condition parser
     Join join = call.rel(0);
     RelNode leftInput = join.getInput(0);
     RelNode rightInput = join.getInput(1);
 
-    RelNode leftExchange = LogicalExchange.create(leftInput, RelDistributions.SINGLETON);
-    RelNode rightExchange = LogicalExchange.create(rightInput, RelDistributions.BROADCAST_DISTRIBUTED);
+    RelNode leftExchange;
+    RelNode rightExchange;
+    List<RelHint> hints = join.getHints();
+    if (hints.contains(PinotRelationalHints.USE_HASH_JOIN)) {
```

Comment:
Why is this if-else ? In the else part, we will still be doing hashjoin in the current implementation

Reply:
renamed. sorry for the confusion this should be `USE_HASH_DISTRIBUTE`. has nothing to do with the join type